src/Controller/EnMarche/ProcurationController.php

```diff
@@ -27,104 +26,108 @@ class ProcurationController extends Controller
-     * @Method("GET")
-     */
-    public function indexAction(Request $request, ProcurationRequestSession $procurationSession): Response
-    {
-        $procurationSession->start();
-
-        return $this->render('procuration/index.html.twig', [
-            'has_error' => $request->query->getBoolean('has_error'),
-            'form' => $this->createForm(ProcurationVoteType::class, $procurationSession->getCurrentModel())->createView(),
-        ]);
-    }
-
-    /**
-     * @Route("/je-demande/mon-lieu-de-vote", defaults={"_enable_campaign_silence"=true}, name="app_procuration_request_vote")
+     * @Route(
+     *     "/choisir/{action}",
+     *     defaults={"_enable_campaign_silence"=true},
+     *     requirements={"action"="proposition|demande"},
+     *     name="app_procuration_choose_election"
+     * )
      * @Method("GET|POST")
      */
-    public function voteAction(Request $request, ProcurationRequestSession $procurationSession): Response
+    public function chooseElection(Request $request, string $action, ProcurationSession $procurationSession): Response
     {
-        $form = $this->createForm(ProcurationVoteType::class, $procurationSession->getCurrentModel())
+        $form = $this->createForm(ElectionContextType::class)
             ->handleRequest($request)
         ;
 
         if ($form->isSubmitted() && $form->isValid()) {
-            return $this->redirectToRoute('app_procuration_request_profile');
+            $procurationSession->setElectionContext($form->getData());
+
+            if ('demande' === $action) {
+                return $this->redirectToRoute('app_procuration_request', ['step' => ProcurationRequest::STEP_URI_VOTE]);
+            }
+
+            return $this->redirectToRoute('app_procuration_proxy_proposal');
         }
 
-        return $this->render('procuration/vote.html.twig', [
-            'form' => $form->createView(),
+        return $this->render('procuration/choose_election.html.twig', [
+            'action' => $action,
+            'elections_form' => $form->createView(),
         ]);
     }
 
     /**
-     * @Route("/je-demande/mes-coordonnees", defaults={"_enable_campaign_silence"=true}, name="app_procuration_request_profile")
+     * @Route("/je-demande", defaults={"_enable_campaign_silence"=true}, name="app_procuration_index_legacy")
+     * @Route(
+     *     "/je-demande/{step}",
+     *     defaults={"_enable_campaign_silence"=true},
+     *     requirements={"step"="mon-lieu-de-vote|mes-coordonnees|ma-procuration"},
+     *     name="app_procuration_request"
+     * )
      * @Method("GET|POST")
      */
-    public function profileAction(Request $request, ProcurationRequestSession $procurationSession): Response
+    public function request(Request $request, ProcurationSession $procurationSession, ?string $step, string $_route): Response
     {
-        $procurationRequest = $procurationSession->getCurrentModel();
+        if ('app_procuration_index_legacy' === $_route) {
```

Comment:
can this be done direclty in YAML ? https://symfony.com/doc/current/routing/redirect_in_config.html ?

Reply:
I've thought about it, but It will be definitely easier to handle the concurrency by defining it here.

Reply:
concurrency ?

Reply:
Which one is matched first.

Reply:
just declare the route before.

Reply:
https://github.com/EnMarche/en-marche.fr/blob/master/app/config/routing.yml

Reply:
We usually define legacy urls after for performance reason, what about this consistency?

Reply:
redirecting legacy ?

Reply:
yup.

Reply:
okay then